src/common/backport14.h

```diff
@@ -21,18 +21,119 @@
 // Library code from C++14 that can be implemented in C++11.
 
 namespace ceph {
-template<typename T>
-using remove_extent_t = typename std::remove_extent<T>::type;
-template<typename T>
-using remove_reference_t = typename std::remove_reference<T>::type;
-template<typename T>
-using result_of_t = typename std::result_of<T>::type;
+// Template variable aliases from C++17 <type_traits>
```

Comment:
can we guard this with `#if __cplusplus < 201703L`, and just put these templates into `namespace std`?